### ISSUE TITLE
IndexOutOfBounds crash

### ISSUE THREAD

**Issue by Amcio** [maksim789456/DynamicSurroundingsFabric#29]
My game crashed with the following error:
```java
Error executing task on Sound executor
java.lang.ArrayIndexOutOfBoundsException: Index 255 out of bounds for length 255
	at org.orecruncher.dsurround.runtime.audio.SoundFXProcessor.lambda$onSoundPlay$1(SoundFXProcessor.java:146) ~[DynamicSurroundings-RemasteredFabric-1.19-0.1.1.jar:?]
	at net.minecraft.class_4235$class_4236.method_19737(class_4235.java:34) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:157) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:131) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18857(class_1255.java:140) ~[client-intermediary.jar:?]
	at net.minecraft.class_4238.method_19765(class_4238.java:42) ~[client-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:1589) [?:?]
Allocate new source: Unable to allocate memory.
Allocate new source: Unable to allocate memory.
Allocate new source: Unable to allocate memory.
Allocate new source: Unable to allocate memory.
```
It crashed right after I entered an area with a lot of mobs, might be related.

**Comment by maksim789456** [comment]
I tried to simulate this by spawn a ~350 mob sounds effects, but I don't catch the crash
But I add check to prevent such cases in next version

**Comment by Alkiasan** [comment]
Having this issue too, is the fix for this going to be pushed anytime soon?

**Comment by maksim789456** [comment]
Fixed at #31
